spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebAutoConfiguration.java

```diff
@@ -65,12 +66,17 @@ public TraceWebAspect traceWebAspect() {
 	}
 
 	@Bean
-	public FilterRegistrationBean traceWebFilter(ApplicationEventPublisher publisher) {
+	@ConditionalOnMissingBean
+	public TraceFilter traceFilter(ApplicationEventPublisher publisher) {
 		Pattern pattern = StringUtils.hasText(this.skipPattern) ? Pattern.compile(this.skipPattern)
```

Comment:
Not positive, but you might get double registrations as boot automatically registers filters. @dsyer could day for sure.